pkg/ddl/executor.go

```diff
@@ -1382,21 +1382,21 @@ func preSplitAndScatter(ctx sessionctx.Context, store kv.Storage, tbInfo *model.
 		return
 	}
 	var (
-		preSplit      func()
-		scatterRegion bool
+		preSplit func()
+		scope    scatterScope
 	)
-	val, err := ctx.GetSessionVars().GetGlobalSystemVar(context.Background(), variable.TiDBScatterRegion)
-	if err != nil {
-		logutil.DDLLogger().Warn("won't scatter region", zap.Error(err))
+	val, ok := ctx.GetSessionVars().GetSystemVar(variable.TiDBScatterRegion)
+	if !ok {
+		logutil.DDLLogger().Warn("won't scatter region")
 	} else {
-		scatterRegion = variable.TiDBOptOn(val)
+		scope = scatterScope(val)
 	}
 	if len(parts) > 0 {
-		preSplit = func() { splitPartitionTableRegion(ctx, sp, tbInfo, parts, scatterRegion) }
+		preSplit = func() { splitPartitionTableRegion(ctx, sp, tbInfo, parts, scope) }
 	} else {
-		preSplit = func() { splitTableRegion(ctx, sp, tbInfo, scatterRegion) }
+		preSplit = func() { splitTableRegion(ctx, sp, tbInfo, scope) }
 	}
-	if scatterRegion {
+	if scope != scatterOff {
```

Comment:
Why here "scope == scatteOff" we need start new go routine for preSplit?

Reply:
When `scope != scatterOff `, the `CREATE TABLE` statement need to wait all split/scatter region finish. IOW, `scope == scatteOff` indicate split region will not block the `CREATE TABLE` statement return.

> the `CREATE TABLE` statement returns success only after the Regions are successfully scattered.
refer: https://docs.pingcap.com/tidb/stable/system-variables#tidb_scatter_region